DuckDuckGo/TabViewController.swift

```diff
@@ -952,21 +959,22 @@ class TabViewController: UIViewController {
         } else {
             privacyInfo = nil
         }
-        
         onPrivacyInfoChanged()
     }
     
     public func makePrivacyInfo(url: URL) -> PrivacyInfo? {
         guard let host = url.host else { return nil }
-        
+
         let entity = ContentBlocking.shared.trackerDataManager.trackerData.findEntity(forHost: host)
-        
-        
         let privacyInfo = PrivacyInfo(url: url,
                                       parentEntity: entity,
                                       protectionStatus: makeProtectionStatus(for: host))
-        privacyInfo.serverTrust = webView.serverTrust
-        
+        let isValid = certificateTrustEvaluator.evaluateCertificateTrust(trust: webView.serverTrust)
+        if let isValid {
+            privacyInfo.serverTrust = isValid ? webView.serverTrust : nil
+        }
+        privacyInfo.isSpecialErrorPageVisible = (isValid == nil)
+
```

Comment:
This part should be easy to test since we inject the certificateTrustEvaluator and you can mock it.nIf you look at TabViewControllerDaxDialogTests you can see how the TabViewController can be initialised for test